Firebase/Core/FIRApp.m

```diff
@@ -293,26 +301,39 @@ - (instancetype)initInstanceWithName:(NSString *)name options:(FIROptions *)opti
 - (BOOL)configureCore {
   [self checkExpectedBundleID];
   if (![self isAppIDValid]) {
-    if (_options.usingOptionsFromDefaultPlist && [self isDataCollectionDefaultEnabled]) {
+    if (_options.usingOptionsFromDefaultPlist && [self isDataCollectionDefaultEnabled] &&
```

Comment:
It's strange that we both check for isDataCollectionDefaultEnabled and pass it in

Reply:
I agree, in the connector it's just set to YES, and I didn't write the original code.